akka-actor-typed/src/main/scala/akka/actor/typed/internal/routing/RoutingLogic.scala

```diff
@@ -76,14 +78,51 @@ private[akka] object RoutingLogics {
 
     private var currentRoutees: Array[ActorRef[T]] = _
 
-    override def selectRoutee(): ActorRef[T] = {
+    override def selectRoutee(msg: T): ActorRef[T] = {
       val selectedIdx = ThreadLocalRandom.current().nextInt(currentRoutees.length)
       currentRoutees(selectedIdx)
     }
+
     override def routeesUpdated(newRoutees: Set[ActorRef[T]]): Unit = {
       currentRoutees = newRoutees.toArray
     }
 
   }
 
+  final class ConsistentHashingLogic[T](
+      virtualNodesFactor: Int = 0,
+      mapping: ConsistentHashMapping[T] = ConsistentHashingLogic.emptyHashMapping)
+      extends RoutingLogic[T] {
+
+    private var currentRoutees: Set[ActorRef[T]] = _
+
+    private var currentConsistentHash: ConsistentHash[ActorRef[T]] = ConsistentHash(currentRoutees, virtualNodesFactor)
+
+    override def selectRoutee(msg: T): ActorRef[T] = {
+      mapping(msg) match {
+        case bytes: Array[Byte] => currentConsistentHash.nodeFor(bytes)
+        case str: String        => currentConsistentHash.nodeFor(str)
+        case x: AnyRef          => ??? // What serialization mechanism we should use here?
```

Comment:
In non-typed akka following piece of code is responsible for serialization of `AnyRef`: 
```
SerializationExtension(system).serialize(x).get
```
What approach should I take here?